### ISSUE TITLE
How to populate input variable length data

### ISSUE THREAD

**Issue by zyk19981118** [lucidrains/se3-transformer-pytorch#6]
Hello, thank you for your work.
I am using the implementation of your SE3-Transformer as part of my project, but I have encountered some problems when processing variable length data input into your model. I do not know how to fill in features, coordinates and masks to meet the needs of the model.

The processing of input data in my DataSet is as follows:
![image](https://user-images.githubusercontent.com/53036974/110742889-002ac500-81ec-11eb-96d1-659caea8b96e.png)
I fill the coordinates and features with zeros to ensure that their input dimensions are (246,3) and (246,20) respectively, and fill the mask with true and false bool types.
In this case,there are 49 valid values,.So the shapes with valid features and coordinates are (49,20) and (49,3) and the rest are padded to zero. The first 49 of the mask is true and the rest is false

Then I check through the torch.utils.data.DataLoader get input dimension is no problem:
![image](https://user-images.githubusercontent.com/53036974/110742911-0b7df080-81ec-11eb-97f9-0f9bfdfd8b19.png)
My network looks like this, A typical binary classifier，I output the result of se3：
 
![image](https://user-images.githubusercontent.com/53036974/110742941-1cc6fd00-81ec-11eb-9b32-c4966d40cd64.png)

However, the output of se3 is NaN as shown in the figure below.
![image](https://user-images.githubusercontent.com/53036974/110742993-2f413680-81ec-11eb-924c-f5a6eb8a1562.png)
The first 48 have results, the rest are NaN, not clear what the problem is

The next batch may all be NaN due to model weight changes
![image](https://user-images.githubusercontent.com/53036974/110743017-37997180-81ec-11eb-8b9d-366c87bdf782.png)
The output of SE3 causes all my losses to be NaN

I think these NaN are probably caused by my incorrect filling method. I have also tried to fill in "1", but it was also ineffective. Maybe the mask I typed in can't have false, and I'm confused about that. Could you please tell me how to correctly fill coordinates, features and masks or give me some advice on using your SE3 model to handle variable length data.
Hope this helps! ! thank you

**Comment by lucidrains** [comment]
Hi! Could you possibly provide me the full script where I can reproduce the NaN error?

**Comment by zyk19981118** [comment]
Thanks, I've written a simple README that I hope will be helpful in reproducing the error
[test_se3.zip](https://github.com/lucidrains/se3-transformer-pytorch/files/6127472/test_se3.zip)

**Comment by lucidrains** [comment]
@zyk19981118 thank you for the scripts! I believe I have fixed the issue in `0.0.23` https://github.com/lucidrains/se3-transformer-pytorch/releases/tag/0.0.23

**Comment by lucidrains** [comment]
@zyk19981118 also welcome you to try https://github.com/lucidrains/egnn-pytorch or https://github.com/lucidrains/En-transformer

**Comment by zyk19981118** [comment]
thanks a lot！

**Comment by lucidrains** [comment]
@zyk19981118 I also took a look at your custom network, and decided to add a feature, where you can do `net(feats, coors, mask = mask, return_pooled = True)`

by setting `returned_pooled = True`, it will do a masked mean over your sequence, which you can then linear out for your prediction

**Comment by zyk19981118** [comment]
That's a good idea，waiting for your good news.

**Comment by lucidrains** [comment]
It's already added, go try it out!
